### PR TITLE
Anti debugging review

### PR DESCRIPTION
**Pull request by cpholguera**
## Help for reviewers

The _real_ content to review is in:
- Document/0x05c-Reverse-Engineering-and-Tampering.md
- Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md
- Document/0x06c-Reverse-Engineering-and-Tampering.md
- Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md

The following contain only minor changes/corrections:
- Document/0x04c-Tampering-and-Reverse-Engineering.md
- Document/0x05a-Platform-Overview.md
- Document/0x05d-Testing-Data-Storage.md
- Document/0x06a-Platform-Overview.md

More info in the commit messages.

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit 53ce271d568db849848b3b4e9137f9c8f09f3502: Broken link result: [!] Error(s) found by the Links validator: 23 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       46 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 674dddce0a3a889e8be4a86ea65830e770fac469: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       49 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 2c98e4e47245f573f4850ebe8a40a3324f08fad8: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       49 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 0a4964c3394514934e2b6a4c6ab6a0a988b90438: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       49 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 59bd399b1e2f82fe3f25ec0cac7ce89a6ea92512: Broken link result: [!] Error(s) found by the Links validator: 23 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       22 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 419284c571e3eb33a6160532c5d3db7b06b19eb7: Broken link result: [!] Error(s) found by the Links validator: 23 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       22 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 3a4b0c73262ed30f684eac0b88c4142d28d6dd73: Broken link result: [!] Error(s) found by the Links validator: 23 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       22 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 9c77c96db665e8ef9ed3c8d23cb455407374d771: Broken link result: [!] Error(s) found by the Links validator: 24 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       26 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 1]
```diff
@@ -393,46 +393,32 @@ JNIEXPORT void JNICALL Java_sg_vantagepoint_jdwptest_MainActivity_JDWPfun(
...
-                if (line.length() > tracerpid.length()) {
-                    if (line.substring(0, tracerpid.length()).equalsIgnoreCase(tracerpid)) {
-                        if (Integer.decode(line.substring(tracerpid.length() + 1).trim()) > 0) {
-                            return true;
-                        }
-                        break;
-                    }
-                }
-            }
+This technique is usually applied within the JNI native libraries in C, you can take a look at [Google's gperftools (Google Performance Tools)) Heap Checker](https://github.com/gperftools/gperftools/blob/master/src/heap-checker.cc#L112 "heap-checker.cc - IsDebuggerAttached") implementation of the `IsDebuggerAttached` method. However, if you prefer to include this check as part of your Java/Kotlin code you can refer to this Java implementation of the `hasTracerPid` method from [Tim Strazzere's Anti-Emulator project](https://github.com/strazzere/anti-emulator/ "anti-emulator").
```
Comment (TheDauntless) [inline Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md:406]:
```suggestion
This technique is usually applied within the JNI native libraries in C, as shown in [Google's gperftools (Google Performance Tools)) Heap Checker](https://github.com/gperftools/gperftools/blob/master/src/heap-checker.cc#L112 "heap-checker.cc - IsDebuggerAttached") implementation of the `IsDebuggerAttached` method. However, if you prefer to include this check as part of your Java/Kotlin code you can refer to this Java implementation of the `hasTracerPid` method from [Tim Strazzere's Anti-Emulator project](https://github.com/strazzere/anti-emulator/ "anti-emulator").

```

File: Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 2]
```diff
@@ -942,18 +928,18 @@ Work on bypassing the defenses and answer the following questions:
 - Did you need to write custom code to disable the anti-emulation feature(s)? How much time did you need?
 - What is your assessment of the difficulty of bypassing the mechanisms?
 
-### Testing Run Time Integrity Checks (MSTG-RESILIENCE-6)
+### Testing Runtime Integrity Checks (MSTG-RESILIENCE-6)
 
 #### Overview
 
-Controls in this category verify the integrity of the app's memory space to defend the app against memory patches applied during run time. Such patches include unwanted changes to binary code, byte-code, function pointer tables, and important data structures, as well as rogue code loaded into process memory. Integrity can be verified by
+Controls in this category verify the integrity of the app's memory space to defend the app against memory patches applied during runtime. Such patches include unwanted changes to binary code, byte-code, function pointer tables, and important data structures, as well as rogue code loaded into process memory. Integrity can be verified by
```
Comment (TheDauntless) [inline Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md:935]:
```suggestion
Controls in this category verify the integrity of the app's memory space to defend the app against memory patches applied during runtime. Such patches include unwanted changes to binary code, byte-code, function pointer tables, and important data structures, as well as rogue code loaded into process memory. Integrity can be verified by:

```

Comment (TheDauntless) [reply Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md:935]: Maybe the list below this line should be a list though, as it's written as a long sentence anyway.

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 3]
```diff
@@ -393,17 +393,17 @@ launchd   1 root   42u  IPv4 0x69c2ce211253b90b      0t0  TCP 192.168.1.12:ssh->
...
 
-The XNU kernel implements the `ptrace` system call, but some of the call's functionality (including reading and writing register states and memory contents) has been eliminated. Nevertheless, `ptrace` is used in limited ways by standard debuggers, such as LLDB and GDB. Some debuggers, including Radare2's iOS debugger, don't invoke `ptrace` at all.
+> For more information you can refer to the LLVM project in GitHub which contains the [source code for LLDB](https://github.com/llvm/llvm-project/tree/master/lldb "LLDB") as well as Chapter 5 "Non Sequitur: Process Tracing and Debugging" and 13 "BS"D - The BSD Layer" from "Mac OS X and iOS Internals: To the Apple's Core" [#levin] and Chapter 4 "Tracing and Debugging" from "The Mac Hacker's Handbook" [#miller].
```
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md:402]: the section is called literally **BS"D - The BSD Layer** (with that double quote), it's pretty weird in fact ^^

**Comment by OMTGreleaser** [comment]
Results for commit 90e60fd00dc5fe532fe9bf87275511bef45c5a45: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       28 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 76731092ee58ad7fbec4d205dba05c0187115ccc: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       29 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x06c-Reverse-Engineering-and-Tampering.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: Actually I think this might be clearer without the chapter names, removed ;)

**Comment by OMTGreleaser** [comment]
Results for commit 1cdf5922839997c4669ace23c1bb85d8a4765d58: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       30 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit bac1251add9740af6c6565478f991ff0f0bb05d8: Broken link result: [!] Error(s) found by the Links validator: 22 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       30 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Pull request merged**